packages/abstract-core/src/objects/gov_type.rs

```diff
@@ -1,54 +1,141 @@
 //! # Governance structure object
 
+use cosmwasm_std::{Addr, Api};
+use cw_address_like::AddressLike;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
+use crate::AbstractError;
+
+const MIN_GOV_TYPE_LENGTH: usize = 4;
+const MAX_GOV_TYPE_LENGTH: usize = 64;
+
 /// Governance types
 #[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
-
-pub enum GovernanceDetails {
+#[non_exhaustive]
```

Comment:
Good call on this. Would an example for additional "hard-coded" governance type be DaoDao or Vectis for example?

Reply:
could be, or `SelfOwned` in the case where we set the proxy as the owner! 
Account inception...